deployment/puppet/cluster/files/q-agent-cleanup.py

```diff
@@ -477,6 +488,46 @@ def _reschedule_agent_l3(self, agent_type):
         self.log.info("_reschedule_agent_l3: ended rescheduling of orphaned routers")
         self.log.debug("_reschedule_agent_l3: end.")
 
+    def _reschedule_agent_lbaas(self, agent_type):
+        self.log.debug("_reschedule_agent_lbaas: start.")
+        alive_agents = []
+
+        # collect pool from removed lbaas agent
+        remove_pools = []
+        dead_agent = None
+        agent_host = socket.gethostname()
```

Comment:
恩， 有一点没有想通，如果是本机自己在ocf 的stop action中调用这个q-agent-cleanup.py 进行迁移那么 根本迁移不出去。

还有就是在 l3 或者dhcp ocf脚本都是在 start的最后调用的这个q-agent-cleanup.py中的schedule，
它们的原意实在只有一个l3 agent或者dhcp agent
的时候，才进行相关的reschedule

2016-07-19 16:08 GMT+08:00 huntxu <notifications@github.com>:

> In deployment/puppet/cluster/files/q-agent-cleanup.py
> <https://github.com/eayunstack/fuel-library/pull/47#discussion_r71289245>:
>
> > @@ -477,6 +488,46 @@ def _reschedule_agent_l3(self, agent_type):
> >          self.log.info("_reschedule_agent_l3: ended rescheduling of orphaned routers")
> >          self.log.debug("_reschedule_agent_l3: end.")
> >
> > +    def _reschedule_agent_lbaas(self, agent_type):
> > +        self.log.debug("_reschedule_agent_lbaas: start.")
> > +        alive_agents = []
> > +
> > +        # collect pool from removed lbaas agent
> > +        remove_pools = []
> > +        dead_agent = None
> > +        agent_host = socket.gethostname()
>
>
> 这个脚本不是简单的把本机上运行着的lb-agent认为已死然后操作，而是可以通用的在一台第三方机器执行，所以建议参考上面l3和dhcp的重新调度过程改写，符合通用性
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eayunstack/fuel-library/pull/47/files/34227f5f94dbd51594b544910280d08652a6d3d8#r71289245>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGUT0adXwKZJvnNUg7Gy3XQG265maNucks5qXIYBgaJpZM4JOgK3>
> .
>